### ISSUE TITLE
Splitting metrics & losses implementations

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras-core#12]
At this time it seems that @ariG23498, @grasskin, and @AakashKumarNain are working on losses or metrics, which can lead to some collisions. Let's split the work:

@ariG23498:

- [Accuracy metrics](https://keras.io/api/metrics/accuracy_metrics/)

@grasskin:

- [Regression metrics](https://keras.io/api/metrics/regression_metrics/)
- [Regression losses](https://keras.io/api/losses/regression_losses/)

@AakashKumarNain:

- [Hinge losses](https://keras.io/api/losses/hinge_losses/)
- [Hinge metrics](https://keras.io/api/metrics/hinge_metrics/)

This includes both plain functions and their wrapper classes. All objects should have unit tests and should have a complete docstring (for the most part you can just copy the Keras docstring and remove the TF references, etc.).

For loss functions that are redundant with metrics, the plain function should live in `losses/` and be imported in `metrics/`. The class-based wrapper should exist both as a Loss in `losses/` and as a Metric in `metrics/`.
